components/chef-run/lib/chef-run/target_host.rb

```diff
@@ -81,12 +81,13 @@ def connect!
     rescue Train::Error => e
       # These are typically wrapper errors for other problems,
       # so we'll prefer to use e.cause over e if available.
-      raise ConnectionFailure.new(e.cause || e, opts)
+      raise ConnectionFailure.new(e.cause || e, config)
     end
 
-    # Returns the user being used to connect
+    # Returns the user being used to connect. Defaults to root if not specified or
+    # defaulted in .ssh/config (for ssh connections), as set up in '#apply_ssh_config'.
     def user
-      config[:user]
+      config[:user] || "root"
```

Comment:
Super minor - does Train export this anywhere. EG, could we do `config[:user] || Train::Connection.default_user` instead of us hardcoding "root"?

Reply:
Good idea,  I think we can. We'll have to create a discardable 'ssh' transport instance to get that value if we don't have it already, but that's no big deal.

Reply:
Updated, looks like default_options was provided at the class-level so we're good without having to instantiate.